libcontainer/container_linux.go

```diff
@@ -1199,7 +1199,7 @@ func (c *linuxContainer) makeCriuRestoreMountpoints(m *configs.Mount) error {
 		if err != nil {
 			return err
 		}
-		if err := checkProcMount(c.config.Rootfs, dest, ""); err != nil {
+		if err := checkProcMount(c.config.Rootfs, dest, "", c.config); err != nil {
```

Comment:
ditto (maybe use singe `c.config` instead of two arguments).

Reply:
or make checkProcMount a method of `linuxContainer`

Reply:
I considered moving `checkProcMount` to `container_linux.go` and making it a method of `linuxContainer` but that doesn't seem to be feasible without moving a bunch of `rootfs_linux.go`, too:

[![](https://mermaid.ink/img/eyJjb2RlIjoiJSUgaHR0cHM6Ly9naXRodWIuY29tL29wZW5jb250YWluZXJzL3J1bmMvcHVsbC8zMjI0XG4lJSBodHRwczovL2dpdGh1Yi5jb20vb3BlbmNvbnRhaW5lcnMvcnVuYy90cmVlL2QxYzliNDNlL2xpYmNvbnRhaW5lclxuZ3JhcGggVERcblxuICBzdWJncmFwaCByb290ZnNfbGludXguZ29cbiAgICBwcmVwYXJlQmluZE1vdW50XG4gICAgY2hlY2tQcm9jTW91bnRcbiAgICBtb3VudFRvUm9vdGZzXG5cbiAgICBwcmVwYXJlQmluZE1vdW50IC0tPiBjaGVja1Byb2NNb3VudFxuICAgIG1vdW50VG9Sb290ZnMgLS0-fGNhc2UgYmluZHwgcHJlcGFyZUJpbmRNb3VudFxuICAgIG1vdW50VG9Sb290ZnMgLS0-fGNhc2UgY2dyb3VwfCBtb3VudENncm91cFYxXG4gICAgbW91bnRUb1Jvb3RmcyAtLT58ZGVmYXVsdCBjYXNlfCBjaGVja1Byb2NNb3VudFxuXG4gICAgcHJlcGFyZVJvb3RmcyAtLT58Zm9yIGVhY2ggbW91bnR8IG1vdW50VG9Sb290ZnNcbiAgICBtb3VudENncm91cFYxIC0tPiBtb3VudFRvUm9vdGZzXG4gICAgbW91bnRDZ3JvdXBWMSAtLT58Zm9yIGVhY2ggYmluZHwgbW91bnRUb1Jvb3Rmc1xuICBlbmRcblxuICBzdWJncmFwaCBjb250YWluZXJfbGludXguZ29cbiAgICBtYWtlQ3JpdVJlc3RvcmVNb3VudHBvaW50c1xuXG4gICAgbWFrZUNyaXVSZXN0b3JlTW91bnRwb2ludHMgLS0-fGNhc2UgYmluZHwgcHJlcGFyZUJpbmRNb3VudFxuICAgIG1ha2VDcml1UmVzdG9yZU1vdW50cG9pbnRzIC0tPnxkZWZhdWx0IGNhc2V8IGNoZWNrUHJvY01vdW50XG4gIGVuZCIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjpmYWxzZX0)](https://mermaid.live/edit#eyJjb2RlIjoiJSUgaHR0cHM6Ly9naXRodWIuY29tL29wZW5jb250YWluZXJzL3J1bmMvcHVsbC8zMjI0XG4lJSBodHRwczovL2dpdGh1Yi5jb20vb3BlbmNvbnRhaW5lcnMvcnVuYy90cmVlL2QxYzliNDNlL2xpYmNvbnRhaW5lclxuZ3JhcGggVERcblxuICBzdWJncmFwaCByb290ZnNfbGludXguZ29cbiAgICBwcmVwYXJlQmluZE1vdW50XG4gICAgY2hlY2tQcm9jTW91bnRcbiAgICBtb3VudFRvUm9vdGZzXG5cbiAgICBwcmVwYXJlQmluZE1vdW50IC0tPiBjaGVja1Byb2NNb3VudFxuICAgIG1vdW50VG9Sb290ZnMgLS0-fGNhc2UgYmluZHwgcHJlcGFyZUJpbmRNb3VudFxuICAgIG1vdW50VG9Sb290ZnMgLS0-fGNhc2UgY2dyb3VwfCBtb3VudENncm91cFYxXG4gICAgbW91bnRUb1Jvb3RmcyAtLT58ZGVmYXVsdCBjYXNlfCBjaGVja1Byb2NNb3VudFxuXG4gICAgcHJlcGFyZVJvb3RmcyAtLT58Zm9yIGVhY2ggbW91bnR8IG1vdW50VG9Sb290ZnNcbiAgICBtb3VudENncm91cFYxIC0tPiBtb3VudFRvUm9vdGZzXG4gICAgbW91bnRDZ3JvdXBWMSAtLT58Zm9yIGVhY2ggYmluZHwgbW91bnRUb1Jvb3Rmc1xuICBlbmRcblxuICBzdWJncmFwaCBjb250YWluZXJfbGludXguZ29cbiAgICBtYWtlQ3JpdVJlc3RvcmVNb3VudHBvaW50c1xuXG4gICAgbWFrZUNyaXVSZXN0b3JlTW91bnRwb2ludHMgLS0-fGNhc2UgYmluZHwgcHJlcGFyZUJpbmRNb3VudFxuICAgIG1ha2VDcml1UmVzdG9yZU1vdW50cG9pbnRzIC0tPnxkZWZhdWx0IGNhc2V8IGNoZWNrUHJvY01vdW50XG4gIGVuZCIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)

Reply:
Yeah, it's a mess :(

If you can figure out a good way to refactor it, I'm all for it.